app/src/main/java/chat/rocket/android/widget/pageIndicator/PageIndicator.kt

```diff
@@ -0,0 +1,239 @@
+
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
+        init(attrs)
+    }
+
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {
+        init(attrs)
+    }
+
+    private fun init(attrs: AttributeSet?) {
+
+        if (isInEditMode) {
+            circleCount = 3
+        }
+
+        circleRadius = resources.getDimension(R.dimen.default_circle_radius).toInt()
+        circlePadding = resources.getDimension(R.dimen.default_circle_padding).toInt()
+
+        var inactiveColor = ContextCompat.getColor(context, R.color.pageIndicatorInactive)
+        var activeColor = ContextCompat.getColor(context, R.color.colorPrimary)
+        val gravity = Gravity.CENTER
+
+        if (attrs != null) {
+            val typedArray = context.theme.obtainStyledAttributes(attrs, R.styleable.PageIndicator, 0, 0)
+            circleRadius = typedArray.getDimension(R.styleable.PageIndicator_indicatorRadius, circleRadius.toFloat()).toInt()
+            circlePadding = typedArray.getDimension(R.styleable.PageIndicator_indicatorPadding, circlePadding.toFloat()).toInt()
+            inactiveColor = typedArray.getColor(R.styleable.PageIndicator_indicatorInActiveColor, inactiveColor)
+            activeColor = typedArray.getColor(R.styleable.PageIndicator_indicatorActiveColor, activeColor)
+            mGravity = typedArray.getInt(R.styleable.PageIndicator_android_gravity, gravity)
+        }
+
+        activePaint = Paint()
+        activePaint!!.color = activeColor
+
+        inActivePaint = Paint()
+        inActivePaint!!.color = inactiveColor
+
+    }
+
+    fun initViewPager(viewPager: ViewPager) {
+        if (mViewPager === viewPager)
+            return
+
+        if (mViewPager != null)
+            viewPager.addOnPageChangeListener(this)
+
+        if (viewPager.adapter == null)
+            throw IllegalStateException("ViewPager doesn't have an adapter isntance.")
+
+        mViewPager = viewPager
+        mViewPager!!.addOnPageChangeListener(this)
+
+        circleCount = viewPager.adapter!!.count
+        mCurrentDragPage = viewPager.currentItem
+        invalidate()
+    }
+
+    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
+
+        val height = View.resolveSize(desiredHeight, heightMeasureSpec)
+        val width = View.resolveSize(desiredWidth, widthMeasureSpec)
+
+        setMeasuredDimension(width, height)
+    }
+
+    override fun onDraw(canvas: Canvas) {
+
+        // draw behind circles
+        for (i in 0 until circleCount) {
+            val center = getCircleCenter(i)
+            canvas.drawCircle(center, (paddingTop + circleRadius).toFloat(), circleRadius.toFloat(), inActivePaint!!)
+        }
+
+        drawRect(canvas)
+
+    }
+
+    private fun drawRect(canvas: Canvas) {
+
+        if (mViewPager == null)
```

Comment:
You can use a single if statement here, `if (mViewPager == null || mViewPager!!.adapter == null || mViewPager!!.adapter!!.count == 0)`